frontend/app/components/Sidebar.tsx

```diff
@@ -1,42 +1,62 @@
 "use client";
 
 import {
-  Icon,
   Button,
   Link as ChakraLink,
   Divider,
   Flex,
+  Icon,
   IconButton,
+  Menu,
+  MenuButton,
+  MenuItem,
+  MenuList,
   Spacer,
   Text,
-  Tooltip,
   VStack,
+  useDisclosure,
 } from "@chakra-ui/react";
-import { PencilSquareIcon, TrashIcon } from "@heroicons/react/24/outline";
+import React from "react";
+import axios from "axios";
+import {
+  ArrowTopRightOnSquareIcon,
+  EllipsisVerticalIcon,
+  PencilSquareIcon,
+  TrashIcon,
+} from "@heroicons/react/24/outline";
 import { useRouter } from "next/navigation";
+import { useMutation } from "@tanstack/react-query";
 import { useDeleteExtractor, useGetExtractors } from "../utils/api";
-
-const NewIconImported = () => {
-  return <Icon as={PencilSquareIcon} />;
-};
-
-const TrashIconImported = () => {
-  return <Icon as={TrashIcon} />;
-};
+import { getBaseApiUrl } from "../utils/api_url";
+import { ShareModal } from "./ShareModal";
 
 export function Sidebar() {
+  const [shareUUID, setShareUUID] = React.useState("");
+
+  const { isOpen, onClose, onOpen } = useDisclosure();
   const { push } = useRouter();
   const { data } = useGetExtractors();
   const deleteExtractor = useDeleteExtractor();
 
+  const baseUrl = getBaseApiUrl();
+  const mutateShare = useMutation({
+    mutationFn: (uuid: string) =>
```

Comment:
Nit, makes it a little more obvious that it's an async func, although it's not actually necessary for the code to work
```suggestion
    mutationFn: async (uuid: string) =>
```